.github/workflows/gradle.yml

```diff
@@ -27,20 +27,29 @@ jobs:
       backend: ${{ steps.filter.outputs.backend }}
       build: ${{ steps.filter.outputs.build }}
       cli: ${{ steps.filter.outputs.cli }}
+      connectors: ${{ steps.filter.outputs.connectors }}
+      db: ${{ steps.filter.outputs.db }}
       frontend: ${{ steps.filter.outputs.frontend }}
     steps:
       - name: Checkout Airbyte
         uses: actions/checkout@v2
       - uses: dorny/paths-filter@v2
         id: filter
         with:
+          # Note, the following glob expression within a filters are ORs.
           filters: |
             backend:
-              - 'airbyte-**/**'
+              - 'airbyte-!(integrations|webapp|webapp-e2e-tests)/**'
```

Comment:
@cgardens I've improved the backend filter so it ignores the non-backend modules. Lmk if I missed something out.

Reply:
It feels like the right way to do this would be to use actual build caching in gradle instead of setting these manual filters. Not for this PR but is that the right North Star?

Reply:
Yes. We do use gradle's caching under the hood. These filters happen at the build-infrastructure level instead of the build level. It helps us avoid running the same gradle tasks X many times on the various build pipelines.

We can push this into gradle. However that still does not give teams build setup isolation from each other.